### PR TITLE
@KafkaClient Header and Headers Argument Enhancements

### PR DESCRIPTION
**Pull request by brianwyka**
## New Features
- Added ability to pass `Collection<org.apache.kafka.common.header.Header>` to `@KafkaClient` interface methods
- Added ability to pass `org.apache.kafka.common.header.Headers` to `@KafkaClient` interface methods

### CODE REVIEW

**Comment by brianwyka** [comment]
I was unable to get the testcontainers working locally, so relying on test failures (before fixing) to be produced by PR here...

**Comment by ctoestreich** [comment]
@graemerocher @brianwyka and I would like these features for supporting our current enterprise initiatives. Any thoughts on this or a timeline?

**Comment by graemerocher** [comment]
@ctoestreich will take a look at this on Monday. Sorry for the delay, prioritising a whole bunch of stuff right now.

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaClientSpec.groovy [hunk 1]
```diff
@@ -36,7 +38,7 @@ class KafkaClientSpec extends Specification {
         MyClient client = ctx.getBean(MyClient)
 
         when:
-        client.sendSync("test", "hello-world")
+        client.sendSync("test", "hello-world", [new RecordHeader("hello", "world".bytes)])
```
Comment (ctoestreich) [inline kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaClientSpec.groovy:41]: @brianwyka seems silly but what does the test for `[new RecordHeader("hello", "world".bytes), null, new RecordHeader("", null), new RecordHeader(null, null)]` produce?

Comment (brianwyka) [reply kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaClientSpec.groovy:41]: @ctoestreich you can add `, null, new RecordHeader("", null)`, and test still passes, however, `new RecordHeader(null, null)` is not valid since `RecordHeader` does not allow `null` keys.

**Comment by graemerocher** [comment]
Thanks for the contribution. I rewrote the logic a bit to improve performance https://github.com/micronaut-projects/micronaut-kafka/commit/a96b588bbfa90151f2ebcf0328f1c1ef1935a7c3
